src/types.md

```diff
@@ -84,6 +86,13 @@ unsigned bytes holding a sequence of UTF-8 code points. Since `str` is a
 [dynamically sized type], it is not a _first-class_ type, but can only be
 instantiated through a pointer type, such as `&str`.
 
+## Never type
+
+The never type `!` is a type with no values, representing the result of
+computations that never complete. Although it has size `0` the empty bit
+pattern is not a valid representation. Expressions of type `!` can be coerced
+into any other type.
```

Comment:
I think type-coercions.md should be updated too?

Reply:
Thanks! Done.